packages/remix-dev/config/routesConvention.ts

```diff
@@ -49,6 +49,10 @@ export function defineConventionalRoutes(
   });
 
   let routeIds = Object.keys(files).sort(byLongestFirst);
+  let parentRouteIds: { [routeId: string]: string | undefined } = {};
+  for (let id of routeIds) {
+    parentRouteIds[id] = findParentRouteId(routeIds, id);
+  }
```

Comment:
It would give the same result, but is there a reason you went for mutating an object instead of using a built-in array function like `reduce`?

Also: type could be `Record<string, string | undefined>`

```suggestion
  let parentRouteIds = routeIds.reduce<Record<string, string | undefined>>(
    (parentRouteIds, id) => ({
      ...parentRouteIds,
      [id]: findParentRouteId(routeIds, id),
    }),
    {}
  );
```

Reply:
I had seen some other code in remix-dev when working on that PR that used the for..or / push approach so went with that, e.g. https://github.com/remix-run/remix/blob/606bc8229b200b1eb8edbf0f2367fbad41ab82df/packages/remix-dev/config/format.ts#L43-L44

I just updated with reduce.